Branch-SDK/src/io/branch/referral/Branch.java

```diff
@@ -2840,22 +2837,26 @@ private void processNextQueueItem() {
 
                 serverSema_.release();
                 if (req != null) {
-                    //All request except Install request need a valid IdentityID
-                    if (!(req instanceof ServerRequestRegisterInstall) && !hasUser()) {
-                        Log.i("BranchSDK", "Branch Error: User session has not been initialized!");
-                        networkCount_ = 0;
-                        handleFailure(requestQueue_.getSize() - 1, BranchError.ERR_NO_SESSION);
-                    }
-                    //All request except open and install need a session to execute
-                    else if (!(req instanceof ServerRequestInitSession) && (!hasSession() || !hasDeviceFingerPrint())) {
-                        networkCount_ = 0;
-                        handleFailure(requestQueue_.getSize() - 1, BranchError.ERR_NO_SESSION);
+                    if (!req.isProcessWaitLockEnabled()) {
+                        // All request except Install request need a valid IdentityID
+                        if (!(req instanceof ServerRequestRegisterInstall) && !hasUser()) {
+                            Log.i("BranchSDK", "Branch Error: User session has not been initialized!");
+                            networkCount_ = 0;
+                            handleFailure(requestQueue_.getSize() - 1, BranchError.ERR_NO_SESSION);
+                        }
+                        //All request except open and install need a session to execute
+                        else if (!(req instanceof ServerRequestInitSession) && (!hasSession() || !hasDeviceFingerPrint())) {
+                            networkCount_ = 0;
+                            handleFailure(requestQueue_.getSize() - 1, BranchError.ERR_NO_SESSION);
+                        } else {
+                            BranchPostTask postTask = new BranchPostTask(req);
+                            postTask.execute();
+                        }
                     } else {
-                        BranchPostTask postTask = new BranchPostTask(req);
-                        postTask.execute();
+                        networkCount_ = 0;
```

Comment:
another flag - I just want to think through potential edge cases where networkCount_ = 0 and the queue is blocked because of Facebook